File: Swift/Euler004.swift
#!/usr/bin/env swift

import Foundation

func main() {
  var biggest = -1
  for a in 100..<1000 {
    for b in a+1..<1000 {
      let product = a * b
      if product > biggest && isPalindrome(number: product) {
        biggest = product
      }
    }
  }
  print(biggest)
}

func isPalindrome(number: Int) -> Bool {
  let asString = String(number)
  return asString == String(asString.reversed())
}

main()

File: Node/Euler004.js
#!/usr/bin/env node

function main() {
  var biggest = -1;
  for (var a = 100; a < 1000; a++) {
    for (var b = a + 1; b < 1000; b++) {
      let product = a * b;
      if (product > biggest && isPalindrome(product)) {
        biggest = product;
      }
    }
  }
  console.log(biggest);
}

function isPalindrome(number) {
  let asString = number.toString();
  let reversed = [...asString].reverse().join("");
  return asString == reversed;
}

if (require.main === module) {
  main();
}
